db/db_impl.cc

```diff
@@ -1686,9 +1683,7 @@ ArenaWrappedDBIter* DBImpl::NewIteratorImpl(const ReadOptions& read_options,
   ArenaWrappedDBIter* db_iter = NewArenaWrappedDbIterator(
       env_, read_options, *cfd->ioptions(), sv->mutable_cf_options, snapshot,
       sv->mutable_cf_options.max_sequential_skip_in_iterations,
-      sv->version_number, read_callback,
-      ((read_options.snapshot != nullptr) ? nullptr : this), cfd, allow_blob,
```

Comment:
This change make the supported check in ` ArenaWrappedDBIter::Refresh()` not returning not supported as expected. This should be fixed. Maybe `allow_refresh` should be passed to false for snapshot case.